### PR TITLE
doc/rados/configuration: use confval directive to define options.

### PR DESCRIPTION
**Pull request by tchaikov**
this changeset includes the 5th and 6th steps:

- [x] common: extract options into yaml. finished in #40731
- [x] add an option to the .yaml.in files for generating the corresponding entries in `legacy_config_opts.h`, and expand `y2c.py` to interpret this and generate `legacy_config_opts.h` accordingly.  #40841
- [x] add a sphinx extension to extract and render the basic and advanced options as part of document. #40890
- [x] use comments in the .yaml.in for the descriptions of some options which don't have desc field yet. #40890
- [x] migrate more options under `doc/rados/configuration` to use the `.confval::` directive. #40897 (still work in progress)
- [x] use "see_also" to reference other options. #40897
- [ ] split the options groups by its readers / service types. so, for example, mds will be linked against the mds options only. the same applies to all daemons and clients

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
should s/conf_val/confval/

File: doc/_ext/ceph_confval.py [hunk 1]
```diff
@@ -107,9 +107,25 @@ def do_plain_num(value: str, typ: str) -> str:
         return str(int(value))
 
 
+def iec_size(value: int) -> str:
+    units = dict(Ei=60,
```
Comment (liewegas) [inline doc/_ext/ceph_confval.py:111]: Is this ``dict`` ordered?

File: doc/rados/configuration/auth-config-ref.rst [hunk 2]
```diff
@@ -190,27 +190,27 @@ stores daemons keyrings inside their data directory. The default keyring
 locations, and the capabilities necessary for the daemon to function, are shown
 below.
```
Comment (liewegas) [inline doc/rados/configuration/auth-config-ref.rst:191]: I wonder if we should drop this section entirely; the caps are out of date (should be based on profiles) and the keyring locations aren't useful information

**Review by liewegas — APPROVED** [review]
looks good!

File: doc/_ext/ceph_confval.py (hunk 1 above)
Comment (tchaikov) [reply doc/_ext/ceph_confval.py:111]: in Python3, dict is now ordered.

File: doc/rados/configuration/auth-config-ref.rst (hunk 2 above)
Comment (tchaikov) [reply doc/rados/configuration/auth-config-ref.rst:191]: will remove it in a follow-up PR.

Comment (tchaikov) [reply doc/rados/configuration/auth-config-ref.rst:191]: #40911

**Pull request merged**